### PR TITLE
Remove unmaintained CI images and update release workflow

### LINKED ISSUE #3142: 使用Docker run.sh时报错，请求的nfpm@v2.3.1是不存在的。
```shell 
IMAGE=centos7   ./ci/dev/image/run.sh yarn
```

```shell 
Step 16/17 : RUN go get github.com/goreleaser/nfpm/cmd/nfpm@v2.3.1
 ---> Running in 7065637228b1
go get github.com/goreleaser/nfpm/cmd/nfpm@v2.3.1: github.com/goreleaser/nfpm/cmd/nfpm@v2.3.1: Get "https://proxy.golang.org/github.com/goreleaser/nfpm/cmd/nfpm/@v/v2.3.1.info": EOF
The command '/bin/sh -c go get github.com/goreleaser/nfpm/cmd/nfpm@v2.3.1' returned a non-zero code: 1
docker: invalid reference format.
See 'docker run --help'.

```
image

这个是在国内被屏蔽了吗？

### PR DESCRIPTION
**Pull request by oxy**
:warning: This needs to be merged before we make a release!

`ci/images/debian10` was removed in a previous PR, but we still used it in our publish flow... this should fix that, and also update our documentation to stop referring to those images.

Closes #3142.

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -12,23 +12,29 @@ jobs:
...
     steps:
-      - uses: actions/checkout@v1
-      - name: Run ./ci/steps/publish-npm.sh
-        uses: ./ci/images/debian10
+      - uses: actions/checkout@v2
+
+      - name: Download npm package
+        uses: actions/download-artifact@v2
         with:
-          args: ./ci/steps/publish-npm.sh
```
Comment (jsjoeio) [inline .github/workflows/publish.yaml:19]: If we're removing this step, can we delete `publish-npm.sh`?

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -12,23 +12,29 @@ jobs:
...
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          name: npm-package
```
Comment (jsjoeio) [inline .github/workflows/publish.yaml:20]: What is this `name` environment variable used for?  (Just curious)

Comment (oxy) [inline .github/workflows/publish.yaml:20]: Took a second look and need to fix this; `npm-package` is produced by the `ci.yaml` workflow, we need to fetch it from there, this won't work.

**Review by jsjoeio — CHANGES_REQUESTED** [review]
Thanks for picking this up so quickly! 

Sorry for deleting that without realizing we used it here 😅

Two comments! But overall looks great!

**Review by oxy — COMMENTED** [review]
I can't request changes on my own PR so don't merge yet, I reviewed it myself and caught something :sweat_smile:

**Review by jsjoeio — APPROVED** [review]
LGTM!

**Pull request merged**